### ISSUE TITLE
Blockly.Variables.allUsedVariables does not return any non-predefined variables ever

### ISSUE THREAD

**Issue by MichaelZawadzki** [LLK/scratch-blocks#837]
### Expected Behavior

When calling Blockly.Variables.allUsedVariables, for example on a workspace, it is expected that all variables used by the workspace are returned in the list. 

### Actual Behavior

This function always returns an empty array (unless there are special predefined variables added).

### Steps to Reproduce

In the vertical workspace add a block to call this function and output the results. I simply hack the "toCode" function to be the following (since there is a button for it):

```javascript
function toCode(lang) {
        var workspaceVars = Blockly.Variables.allUsedVariables(workspace);
        for (var i=0; i<workspaceVars.length; i++) {
          console.log('Workspace variable: ' + workspaceVars[i]);
        }
      }
```

Once this is in place, in the workspace, add a block to the workspace that has a variable. I go to the 'Data' category and create a new variable. Once its created add one of the generated blocks to the workspace, for example the "set VAR to VALUE" block. 

Now hit the button to trigger the hacked code. Notice no output is generated.

### Operating System and Browser

All, reproduced on Mac on Chrome.

### Cause and Solution

Inside the Blockly.Variables.allUsedVariables function there is a coding error on this line:

`if (varName && !varName.toLowerCase() == ignorableName) {`

Here, the ! operator applies to the string, which returns false. False compared to ignorableName always returns false. Hence, no variables make it through that are in the workspace. 

The fix is simply to add parenthesis around the block so that the ! is applied after the string comparison.

**Comment by MichaelZawadzki** [comment]
I see this was merged into develop, and indeed can verify it fixes the issue! What is the process for now closing this out? Is this something I should do (I do see a "close and comment" button below)?

**Comment by navignaw** [comment]
Oops, forgot to close it myself :) But yes, that's the button we want.
